sentry_streams/src/utils.rs

```diff
@@ -1,28 +1,64 @@
+/// Use this wrapper instead of directly using `with_gil()`.
+/// This wrapper automatically emits a warning trace when the time to acquire
+/// the python GIL is higher than some threshold (defaults to 1 second).
+/// The optional warn_threshold value can be supplied in the first position to
+/// change that.
+/// The function syntax is exactly the same as `pyo3::Python::with_gil`.
+///
+///
+/// # Examples
+///
+/// ```
+/// let py_err = ...;
+/// traced_with_gil!(|py| { py_err.print(py) }).unwrap();
+///
+/// let py_err = ...;
+/// traced_with_gil!(Duration::from_secs(10), |py| {
+///     py_err.print(py)
+/// }).unwrap();
+/// ```
+///
+macro_rules! traced_with_gil {
+    ($function: expr) => {
+        traced_with_gil!(std::time::Duration::from_secs(1), $function)
```

Comment:
(beginner question) Is this basically invoking the macro? So it will resolve to the second branch?